### PR TITLE
subscription-2018-03-01-preview

### PR DESCRIPTION
**Pull request by mjfrb**
## Description

---

This checklist is used to make sure that common guidelines for a pull request are followed.
- [ ] Please add REST spec PR link to the SDK PR
- [ ] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md).**
- [ ] **The pull request does not introduce [breaking changes](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/Documentation/breaking-changes.md).**

### [General Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#general-guidelines)
- [ ] Title of the pull request is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/dev/documentation/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#testing-guidelines)
- [ ] Pull request includes test coverage for the included changes.

### [SDK Generation Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#sdk-generation-guidelines)
- [ ] If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.
- [ ] The generate.cmd file for the SDK has been updated with the version of AutoRest, as well as the commitid of your swagger spec or link to the swagger spec, used to generate the code.
- [ ] The `*.csproj` and `AssemblyInfo.cs` files have been updated with the new version of the SDK.

### CODE REVIEW

File: src/SDKs/Subscription/Management.Subscription/Generated/Models/Page1.cs [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+    using Newtonsoft.Json;
+    using System.Collections;
+    using System.Collections.Generic;
+
+    /// <summary>
+    /// Defines a page in Azure responses.
+    /// </summary>
+    /// <typeparam name="T">Type of the page content items</typeparam>
+    [JsonObject]
+    public class Page1<T> : IPage<T>
```
Comment (wilcobmsft) [inline src/SDKs/Subscription/Management.Subscription/Generated/Models/Page1.cs]: anything we can do about this?

Comment (wilcobmsft) [reply src/SDKs/Subscription/Management.Subscription/Generated/Models/Page1.cs]: If not, no worries. Looks like several other SDKs have the same thing going on.

**Comment by wilcobmsft** [comment]
Looks like the following file is missing from the PR: /src/SDKs/_metadata/subscription_resource-manager.txt

**Review by wilcobmsft — APPROVED** [review]

File: src/SDKs/Subscription/Management.Subscription/Generated/Models/Page1.cs (hunk 1 above)
Comment (mjfrb) [reply src/SDKs/Subscription/Management.Subscription/Generated/Models/Page1.cs]: this is expected

**Review by shahabhijeet — CHANGES_REQUESTED** [review]
If you are planning to publish your nuget package you will need the following:
1) Update nuget version
2) Update AssemblyFileVersion (Update AssemblyVersion if major version bump)
2) Add PackageReleaseNotes

**Comment by dsgouda** [comment]
Please link the PR for the related rest spec 
Looks like there are some breaking changes, please bump the package version in .csproj to `1.1.0-preview`
Pull down the latest changes from upstream and run generate.ps1 again, commit all changes

**Comment by dsgouda** [comment]
@mjfrb we still need REST spec api PR from which the code is being generated here

**Comment by wilcobmsft** [comment]
@dsgouda da https://github.com/Azure/azure-rest-api-specs/pull/2535

**Comment by dsgouda** [comment]
Please do a git clean and run `msbuild build.proj`, this should modify the .props file please commit the changes

**Comment by dsgouda** [comment]
@shahabhijeet Please approve and merge the changes
LGTM

**Review by dsgouda — APPROVED** [review]

**Review by shahabhijeet — APPROVED** [review]

**Pull request merged**